src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/DoNotPassTypesByReferenceTests.cs

```diff
@@ -0,0 +1,171 @@
+{
+    public override void Method1(ref string s) // ... but not here
+    {
+    }
+}",
+                GetCA1045CSharpResultAt(4, 44, "s"));
+
+            await VerifyVB.VerifyAnalyzerAsync(@"
+Public Class BaseClass
+    Public Overridable Sub Method1(ByRef s As String) ' issue here...
+    End Sub
+End Class
+
+Public Class Class1
+    Inherits BaseClass
+
+    Public Overrides Sub Method1(ByRef s As String) ' ... but not here
+    End Sub
+End Class
+",
+                GetCA1045BasicResultAt(3, 42, "s"));
+        }
+
+        [Fact]
+        public async Task CA1045_MethodIsInterfaceImplementation_Diagnostic()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+public interface Interface1
+{
+    void Method1(ref string s); // issue here...
+}
+
+public class Class1 : Interface1
+{
+    public void Method1(ref string s) // ... but not here
+    {
+    }
+}",
+                GetCA1045CSharpResultAt(4, 29, "s"));
+
+            await VerifyVB.VerifyAnalyzerAsync(@"
+Public Interface Interface1
+    Sub Method1(ByRef s As String) ' issue here...
+End Interface
+
+Public Class Class1
+    Implements Interface1
+
+    Public Sub Method1(ByRef s As String) Implements Interface1.Method1 ' ... but not here
+    End Sub
+End Class",
+                GetCA1045BasicResultAt(3, 23, "s"));
+        }
+
+        [Fact]
+        public async Task CA1045_VBNET_ByRefOutParameter_NoDiagnostic()
+        {
+            await VerifyVB.VerifyAnalyzerAsync(@"
+Imports System.Runtime.InteropServices
+
+Public Class Class1
+    Public Sub Method1(s As String, <Out> ByRef c As Class1)
+        c = Nothing
+    End Sub
+End Class");
+        }
+
+        [Fact]
+        public async Task CA1045_PInvokeMethod_Diagnostic()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+using System.Runtime.InteropServices;
+
+public class Class1
+{
+    [DllImport(""Advapi32.dll"", CharSet=CharSet.Auto)]
+    public static extern Boolean FileEncryptionStatus(String filename, ref UInt32 status);
+}",
+                GetCA1045CSharpResultAt(8, 83, "status"));
+
+            await VerifyVB.VerifyAnalyzerAsync(@"
+Imports System.Runtime.InteropServices
+
+Public Class Class1
+    <DllImport(""Advapi32.dll"", CharSet:=CharSet.Auto)>
+    Public Shared Function FileEncryptionStatus(ByVal filename As String, ByRef status As UInteger) As Boolean
+    End Function
+End Class",
+                GetCA1045BasicResultAt(6, 81, "status"));
+        }
+
+        private static DiagnosticResult GetCA1045CSharpResultAt(int line, int column, string parameterName)
+            => VerifyCS.Diagnostic()
+                .WithLocation(line, column)
+                .WithArguments(parameterName);
+
+        private static DiagnosticResult GetCA1045BasicResultAt(int line, int column, string parameterName)
+            => VerifyVB.Diagnostic()
+                .WithLocation(line, column)
+                .WithArguments(parameterName);
```

Comment:
💡 Use `WithSpan` instead of `WithLocation` for these helpers, so the full diagnostic location is clear in the tests.